gridstatus/ercot.py

```diff
@@ -1773,6 +1778,57 @@ def _make_bid_curve(df):
 
         return self.parse_doc(df, verbose=verbose)
 
+    @support_date_range("DAY_START")
+    def get_sced_system_lambda(self, date, verbose=False):
+        """Get System lambda of each successful SCED
+
+        5 Minute Publish Interval
+
+        https://github.com/kmax12/gridstatus/issues/269
+
+        Arguments:
+            date (str, datetime): date to get data for
+            verbose (bool, optional): print verbose output. Defaults to False.
+
+        Returns:
+            pandas.DataFrame: A DataFrame
+
+        """
+
+        # just so we can share parse logic
+        kwargs = dict(
+            report_type_id=SCED_SYSTEM_LAMBDA_RTID,
+            date=date,
+            verbose=verbose,
+            extension="csv",
+        )
+
+        if date == "latest":
+            doc = self._get_document(**kwargs)
+            doc = [doc]
+        else:
+            doc = self._get_documents(**kwargs)
+
+        df = pd.concat([pd.read_csv(i.url, compression="zip") for i in doc])
+
+        def handle_dst(time, dst_flag):
+            # shift an hour forward to get new timezone
+            # shift back to display correct how with new tz
+            if dst_flag == "Y":
+                nt = time + pd.Timedelta(hours=1)
+                nt.dt.tz_localize(self.default_timezone, infer=True)
+                nt = time - pd.Timedelta(hours=1)
+            else:
+                return time
+
+        # ANCHOR - handle dst
+        df["SCEDTimeStamp"] = np.vectorize(handle_dst)(
```

Comment:
i think it's better to avoid numpy here and just use the pandas version, which i think is sometihng like this 

```
df["SCEDTimeStamp"] = df.apply(lambda row: handle_dst(row["SCEDTimeStamp"], row["RepeatedHourFlag"]), axis=1)
```

Reply:
scratch that, i would just do this 

df["SCEDTimeStamp"] = pd.to_datetime(df["SCEDTimeStamp"]).dt.tz_localize(self.default_timezone, ambiguous=df["RepeatedHourFlag"] == "Y")

Reply:
actually, can you check the logic of how the ambiguious param works and the repeated hour flag? 

unclear to me if RepeatedHourFlag is Y or N when it is DST. maybe need to be ambiguous=df["RepeatedHourFlag"] == "N"

here's the documentation: https://pandas.pydata.org/docs/reference/api/pandas.Series.tz_localize.html

Reply:
from the docs

```bool-ndarray where True signifies a DST time, False designates a non-DST time (note that this flag is only applicable for ambiguous times)```

I believe those with `"Y"` are the ambiguous times, but we can also wait to hear back from ERCOT

Reply:
based on our slack discussion, i thought we settled that it is set to Y when the time repeats itself during the transition from DST to Standard Time, which would mean that the value of N represents DST. 

but yes, i have a question into ercot and let's see what they say

Reply:
Using an hourly example, when there is a DST event, you'll have 23 hourly events or 25 hourly events. In the xml, that Y flag denotes the event that is a duplicate based on the timestamps. Your assumption is correct. 

FWIW, I've typically seen this implemented by getting rid of "timezone on write", but rather storing the data as utc and then applying timezone on read. Then, you could just use Unix-epoch milliseconds as a format and store the appropriate timezone info in some sort of metadata. However, there are pros/cons to that as well like what happens with timezones change (they do) or having to do timezone conversions 1000s of times for every historical read. 

I've also seen the records flattened, where there is a single row for each day, with a column for each hourly ending value with the column name being the time sans punctuation. To support DST, in the hourly example, you would have a row with a date column followed by 25 potential hourly columns (0000, 0100, 0200, etc). This look great from a storage efficiency because now you are eliminating a bunch of repetitive data however if you want to read a specific column, you'll end up having to write some query magic and would at a minimum typically be reading a full daily row back.